core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -998,6 +1005,13 @@ public static int getMaxGroupingSets(Session session)
         return session.getSystemProperty(MAX_GROUPING_SETS, Integer.class);
     }
 
+    private static void validateHideInaccesibleColumns(boolean value, boolean defaultValue)
+    {
+        if (defaultValue == true && value == false) {
+            throw new TrinoException(INVALID_SESSION_PROPERTY, "hide_inaccessible_columns cannot be set to false if default value is true");
```

Comment:
It is not much user friendly error message, it is very technical. Maybe:

```
hide_inaccessible_columns cannot be disabled with session property when it was enabled with configuration
```

Also use `HIDE_INACCESSIBLE_COLUMNS` constant

Please make sure you have tests for this.